src/AppBanner/index.tsx

```diff
@@ -10,14 +10,15 @@ import type { TrackClick } from '../utils/tracking';
 
 import { AppStore } from '../assets/app-store';
 import { PlayStore } from '../assets/play-store';
-import { StyleData, selfServeStyles } from '../styles/bannerCommon';
+import { BannerColorStyles } from '../styles/colorData';
+import { selfServeStyles } from '../styles/bannerCommon';
```

Comment:
I think a thing that's been confusing me about the configurable colours is that `AppBanner`, `BannerNewsletter` & friends, which don't support configuring the colours, still call `selfServeStyles`. I think that's because `selfServeStyles` serves two purposes:

1. Takes user provided styles, and integrates these with defaults (if, for example, the user provides invalid values, or if certain colours are missing)
2. Generates CSS from a the colour styles object.

In the case of the banners which don't have configurable colours, we only need the second of those two purposes. This could be done in a follow up PR, but I think it would improve the understandability of the code to de-couple those two things. So instead of `selfServeStyles` taking user provided colours and defaults it just takes colours which are guaranteed to be present and returns the CSS.  It wouldn't have to care about where the colours came from (for `AppBanner` etc they'd be hardcoded, for `StyleableBannnerWithLink` they come from combining user provided colours with defaults). At the top level of `StyleableBannerWithLink` it'd have to do the combining of user provided colours and defaults, using `getColors`, and `selfServeStyles` wouldn't call `getColors`.

Thoughts?

Reply:
My long-term utopia is to have a single epic component, and a single banner component (both styleable, though in the case of the epic limited to eg CTA button colours) which Retention can setup/preview in Storybook. Though the competing vision of being able to use RRCP banner/epic designs from Braze is an even better utopia destination.